### PR TITLE
[build] remove FTD/MTD/RADIO from core config file

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR fixes the compile warning.

### CODE REVIEW

File: src/core/config/mac.h [hunk 1]
```diff
@@ -326,13 +326,14 @@
...
  *
- * This setting configures the CSL transmitter feature in Thread 1.2.
- * This is compulsory for 1.2 FTD.
+ * Define to 1 to enable csl transmitter logic.
  *
  */
-#define OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE \
-    (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2) && (OPENTHREAD_FTD || OPENTHREAD_RADIO)
+#ifndef OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+#define OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE 0
```
Comment (bukepo) [inline src/core/config/mac.h:335]:
```suggestion
#define OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
```

File: src/core/thread/topology.cpp [hunk 2]
```diff
@@ -386,7 +386,7 @@ bool Child::HasIp6Address(const Ip6::Address &aAddress) const
     return retval;
 }
 
-#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+#if (OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE)
```
Comment (bukepo) [inline src/core/thread/topology.cpp:389]: Keep consistent with other conditions.
```suggestion
#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
```

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -2278,7 +2278,7 @@ otError Interpreter::ProcessMasterKey(uint8_t aArgsLength, char *aArgs[])
     return error;
 }
 
-#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
+#if (OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE) && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```
Comment (simonlingoogle) [inline src/cli/cli.cpp:2281]:
```suggestion
#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```

File: src/cli/cli.hpp [hunk 4]
```diff
@@ -393,7 +393,7 @@ class Interpreter
 
     otError ParseLinkMetricsFlags(otLinkMetrics &aLinkMetrics, char *aFlags);
 #endif
-#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
+#if (OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE) && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```
Comment (simonlingoogle) [inline src/cli/cli.hpp:396]:
```suggestion
#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```

File: src/core/api/ip6_api.cpp [hunk 5]
```diff
@@ -258,7 +258,7 @@ otError otIp6SelectSourceAddress(otInstance *aInstance, otMessageInfo *aMessageI
     return error;
 }
 
-#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
+#if (OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE) && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```
Comment (simonlingoogle) [inline src/core/api/ip6_api.cpp:261]:
```suggestion
#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```

File: src/core/thread/mlr_manager.hpp [hunk 6]
```diff
@@ -108,11 +108,11 @@ class MlrManager : public InstanceLocator, private NonCopyable
                                     uint16_t            aOldMlrRegisteredAddressNum);
 #endif
 
-#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
+#if (OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE) && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```
Comment (simonlingoogle) [inline src/core/thread/mlr_manager.hpp:111]:
```suggestion
#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with nits

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**